src/lib/battery/battery.cpp

```diff
@@ -188,35 +188,128 @@ void Battery::sumDischarged(const hrt_abstime &timestamp, float current_a)
 
 void Battery::estimateStateOfCharge(const float voltage_v, const float current_a, const float throttle)
 {
-	// remaining battery capacity based on voltage
-	float cell_voltage = voltage_v / _params.n_cells;
 
-	// correct battery voltage locally for load drop to avoid estimation fluctuations
-	if (_params.r_internal >= 0.f) {
-		cell_voltage += _params.r_internal * current_a;
-
-	} else {
-		// assume linear relation between throttle and voltage drop
-		cell_voltage += throttle * _params.v_load_drop;
+	bool sees_coulomb_counting_only = true;
+	// Sees.ai modification coulomb_counting_only
+	// Set to 'true' for Sees behaviour:
+	// SoC relies solely on mAh discharged.
+	//
+	// Set to 'false' for default PX4 behaviour:
+	// SoC is based on whichever of the two options below that provides the lower value:
+	// - A combination of discharge and Voltage based estimation (relies more on voltage at lower voltages).
+	// - Just on discharge.
+
+
+	// -------------------------------------------
+	// If bool is true, do sees desired behaviour
+	// -------------------------------------------
+
+	if (sees_coulomb_counting_only) {
+
+		// This ensures the cell voltage is derived from the filtered voltage and corrects for any voltage load drop as PX4 uses the raw voltage values.
+
+		cell_voltage_filtered = _voltage_filter_v.getState();
+		cell_voltage_filtered /= _params.n_cells;
+		cell_voltage_filtered_load += _params.r_internal * current_a;
+
+		// Derives initial SoC Estimation from the voltage. A 15 second timer is included to allow voltage to settle whilst components initialise.
+		if(first_run || ((hrt_absolute_time() - first_run_time) < 15'000'000)){
```

Comment:
Instead of running for 15 seconds, we could instead just run this any time that current has been low for a time - that way it'll update if you land and takeoff again.  (Small point, this does work fine as is generally).